include/rmm/device_uvector.hpp

```diff
@@ -119,6 +119,127 @@ class device_uvector {
   {
   }
 
+  /**
+   * @brief Returns pointer to the specified element
+   *
+   * Behavior is undefined if `element_index >= size()`.
+   *
+   * @param element_index Index of the specified element.
+   * @return T* Pointer to the desired element
+   */
+  pointer element_ptr(std::size_t element_index) noexcept
+  {
+    assert(element_index < size());
+    return data() + element_index;
+  }
+
+  /**
+   * @brief Returns pointer to the specified element
+   *
+   * Behavior is undefined if `element_index >= size()`.
+   *
+   * @param element_index Index of the specified element.
+   * @return T* Pointer to the desired element
+   */
+  const_pointer element_ptr(std::size_t element_index) const noexcept
+  {
+    assert(element_index < size());
+    return data() + element_index;
+  }
+
+  /**
+   * @brief Performs a synchronous copy of `v` to the specified element in device memory.
+   *
+   * Because this function synchronizes the stream `s`, it is safe to destroy or modify the object
+   * referenced by `v` after this function has returned.
+   *
+   * @note: This function incurs a host to device memcpy and should be used sparingly.
+   *
+   * Example:
+   * \code{cpp}
+   * rmm::device_uvector<int32_t> vec(100, stream);
+   *
+   * int v{42};
+   *
+   * // Copies 42 to element 0 on `stream` and synchronizes the stream
+   * vec.set_element(0, v, stream);
+   *
+   * // It is safe to destroy or modify `v`
+   * v = 13;
+   * \endcode
+   *
+   *
+   * @throws rmm::out_of_range exception if `element_index >= size()`
+   *
+   * @param element_index Index of the target element
+   * @param v The value to copy to the specified element
+   * @param s The stream on which to perform the copy
+   */
+  void set_element(std::size_t element_index, T const& v, cudaStream_t s)
+  {
+    RMM_EXPECTS(
+      element_index < size(), rmm::out_of_range, "Attempt to access out of bounds element.");
+    RMM_CUDA_TRY(cudaMemcpyAsync(element_ptr(element_index), &v, sizeof(v), cudaMemcpyDefault, s));
+    RMM_CUDA_TRY(cudaStreamSynchronize(s));
+  }
+
+  /**
+   * @brief Performs an *a*synchronous copy of `v` to the specified element in device memory.
```

Comment:
```suggestion
   * @brief Performs an asynchronous copy of `v` to the specified element in device memory.
```

What did FS Fitzgerald say about exclamation points? :) In the same vein, I think the spelling is explicit and clear without the asterisks.

Reply:
fwiw, I was doing this so the rendered Doxygen would bold the *a*, but doesn't look like it worked.

Reply:
Yeah, I know. But I think it makes the un-rendered docs less readable.